### PR TITLE
Basic bindings for File with tests

### LINKED ISSUE #6: Bindings for file and Maven
Create a Java binding for `File`, `FileMode` and `Implementation`. The binding should be as complete as possible with the exception of methods that deal with other NIX classes.

Set up a _minimal_ pom.xml that build the project on Linux and creates all jars.

Write a unit test for `File` and make sure the tests can be run from maven.

### PR DESCRIPTION
**Pull request by sujithvm**
This pull request fixes #6 . @stoewer Can you review this pull request? Thank you

### CODE REVIEW

**Comment by stoewer** [comment]
Since this should be a minimal setup the question arises whether two *.pom file are really necessary.

File: nix/src/main/java/org/gnode/nix/File.java [hunk 1]
```diff
@@ -0,0 +1,201 @@
...
+import org.bytedeco.javacpp.Pointer;
+import org.bytedeco.javacpp.annotation.*;
+
+import java.util.ArrayList;
+
+@Platform(value = "linux",
+        include = {"<nix/File.hpp>"},
+        link = {"nix"},
+        linkpath = {"/usr/local/lib/"},
+        includepath = {"/usr/local/include/"})
```
Comment (stoewer) [inline nix/src/main/java/org/gnode/nix/File.java]: Please find a solution for setting `value`, `linkpath` and `includepath` more in the build conf or via environment variables. Or file an issue for M1 and label it as `high priority`.

File: nix/src/main/java/org/gnode/nix/File.java [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+    }
+
+    private native void allocate();
+
+    /**
+     * Copy constructor.
+     *
+     * @param other The file to copy.
+     */
+    public File(@Const @ByRef File other) {
```
Comment (stoewer) [inline nix/src/main/java/org/gnode/nix/File.java]: Due to reference counting in the NIX library the copy ctor might be harmful. If it is not strictly necessary omit the ctor.

File: nix/src/main/java/org/gnode/nix/File.java [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+    String getLocation();
+
+    /**
+     * Get the creation date of the file.
+     *
+     * @return The creation date of the file.
+     */
+    public native
+    @Cast("time_t")
+    long createdAt();
```
Comment (stoewer) [inline nix/src/main/java/org/gnode/nix/File.java]: Rename to `getCreatedAt()`.

Comment (stoewer) [inline nix/src/main/java/org/gnode/nix/File.java]: Maybe its a good idea to convert `long` to a more java-esque time representation. My favorite would be Java 8 `LocalDataTime` or `DateTime` from  Joda-Time (http://www.joda.org/joda-time/index.html).

File: nix/src/main/java/org/gnode/nix/File.java [hunk 4]
```diff
@@ -0,0 +1,201 @@
...
+
+    /**
+     * Get the date of the last update.
+     *
+     * @return The date of the last update.
+     */
+    public native
+    @Cast("time_t")
+    long updatedAt();
```
Comment (stoewer) [inline nix/src/main/java/org/gnode/nix/File.java]: Same as above.

File: nix/src/main/java/org/gnode/nix/Utils.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
+package org.gnode.nix;
```
Comment (stoewer) [inline nix/src/main/java/org/gnode/nix/Utils.java]: Can you move this to `org.gnode.nix.internal` or something similar. Just to make clear that it is not part of the public api.

File: pom.xml [hunk 6]
```diff
@@ -0,0 +1,337 @@
...
+
+    <profile>
+      <id>platform-dependencies</id>
+      <activation>
+        <property>
+          <name>platform.dependencies</name>
+        </property>
+      </activation>
+      <dependencies>
+        <dependency>
```
Comment (stoewer) [inline pom.xml]: Since this should be a minimal configuration please omit all non-linux items.

**Comment by sujithvm** [comment]
@stoewer Now includePath and linkPath can be specified directly from maven. I have updated build instructions on readme. std=c++ can also be specified as compiler option so javacpp does have to be locally installed. Other suggested changes have been done. Please review the changes when free. Thank you

**Comment by stoewer** [comment]
Perfect :+1: 

Continue with #8 ?

**Pull request merged**